### PR TITLE
Text, Image, and Stack properties can be overridden on different states/conditions

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation

Need to be able to override a subset of component properties on different states and conditions

### Description

These changes are only for the deserialization of new data from the API

Add the concept of "partial components" which are the same as the actual components but everything is optional for overriding properties for different states and conditions

### CODE REVIEW

**Review by JayShortway — COMMENTED** [review]
Let's go! Just questions mostly.

File: Sources/Paywalls/Components/Common/PartialComponent.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    struct ComponentConditions<T: PartialComponent>: PaywallComponentBase {
+
+        let mobileLandscape: T?
+        let tablet: T?
+        let tabletLandscape: T?
+        let desktop: T?
+
+    }
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply Sources/Paywalls/Components/Common/PartialComponent.swift:38]: @JayShortway Let's bring this up in today's team sync 😇

File: RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift [hunk 2]
```diff
@@ -35,10 +35,12 @@ struct ButtonComponentView: View {
...
-        AsyncButton(
-            action: { try await performAction() },
-            label: { StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: self.onDismiss) }
-        )
+        AsyncButton {
+            try await performAction()
+        } label: {
+            // Not passing an onDismiss - nothing in this stack should be able to dismiss
+            StackComponentView(viewModel: viewModel.stackViewModel, onDismiss: {})
+        }
```
[the remark this answers is not in the document]
Comment (joshdholtz) [reply RevenueCatUI/Templates/Components/Button/ButtonComponentView.swift:43]: Correct correct!

**Review by fire-at-will — APPROVED** [review]
👏👏

**Pull request merged**